docs/design/assignment.md

```diff
@@ -0,0 +1,329 @@
+# Assignment
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Overview](#overview)
+-   [Syntax](#syntax)
+-   [Simple assignment semantics](#simple-assignment-semantics)
+-   [Built-in types](#built-in-types)
+-   [Tuples, structs, and classes](#tuples-structs-and-classes)
+-   [Extensibility](#extensibility)
+    -   [Simple assignment](#simple-assignment)
+    -   [Arithmetic](#arithmetic)
+    -   [Bitwise and bit-shift](#bitwise-and-bit-shift)
+    -   [Defaults](#defaults)
+-   [Alternatives considered](#alternatives-considered)
+-   [References](#references)
+
+<!-- tocstop -->
+
+## Overview
+
+Values can be assigned to variables using the `=` operator:
+
+```
+var a: i32 = 5;
+a = 6;
+```
+
+For each binary [arithmetic](expressions/arithmetic.md) or
+[bitwise](expressions/bitwise.md) operator `$`, a corresponding compound
+assignment `$=` is provided that performs the operation in-place:
+
+```
+// Same as `a = a + 1;`
+a += 1;
+// Same as `a = a << 3;`
+a <<= 3;
+```
+
+In addition, increment and decrement operators are provided:
+
+```
+// Same as `a = a + 1;`
+++a;
+// Same as `a = a - 1;`
+--a;
+```
+
+These simple assignment, compound assignment, increment, and decrement operators
+can only be used as complete statements, not as subexpressions of other
+operators, even when parenthesized:
+
+```
+var n: i32;
+// Error, assignment is not permitted as a subexpression.
+if (F() and (n = GetValue()) > 5) {
+}
+```
+
+User-defined types can define the meaning of these operations by
+[implementing an interface](#extensibility) provided as part of the Carbon
+standard library.
+
+## Syntax
+
+The operands of these operators can be any [expression](expressions/README.md).
```

Comment:
Are there any parsing concerns, e.g. how is `a*=b` interpreted? Is that a postfix `*` (as in `i32*`) or `*=`?